src/main/java/me/zeroeightsix/kami/module/modules/player/PacketLogger.kt

```diff
@@ -1,100 +1,305 @@
 package me.zeroeightsix.kami.module.modules.player
 
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.launch
 import me.zeroeightsix.kami.KamiMod
+import me.zeroeightsix.kami.event.events.ConnectionEvent
 import me.zeroeightsix.kami.event.events.PacketEvent
-import me.zeroeightsix.kami.mixin.extension.pitch
-import me.zeroeightsix.kami.mixin.extension.yaw
+import me.zeroeightsix.kami.mixin.extension.*
 import me.zeroeightsix.kami.module.Category
 import me.zeroeightsix.kami.module.Module
-import me.zeroeightsix.kami.util.text.MessageSendHelper
-import me.zeroeightsix.kami.util.threads.defaultScope
+import me.zeroeightsix.kami.util.text.MessageSendHelper.sendChatMessage
+import me.zeroeightsix.kami.util.threads.onMainThread
 import me.zeroeightsix.kami.util.threads.safeListener
-import net.minecraft.network.play.client.CPacketPlayer
-import net.minecraft.network.play.client.CPacketPlayerDigging
+import net.minecraft.network.play.client.*
+import net.minecraft.network.play.server.*
 import net.minecraftforge.fml.common.gameevent.TickEvent
-import java.io.File
+import java.io.*
 import java.text.SimpleDateFormat
 import java.util.*
+import kotlin.ConcurrentModificationException
 
 internal object PacketLogger : Module(
     name = "PacketLogger",
     description = "Logs sent packets to a file",
     category = Category.PLAYER
 ) {
-    private val append by setting("Append", false)
-    private val clear = setting("Clear", false)
+    private val showClientTicks by setting("ShowClientTicks", true)
+    private val logInChat by setting("LogInChat", false)
+    private val packetsFrom by setting("LogPacketsFrom", Type.BOTH)
```

Comment:
I'd prefer this to be called `packetType` or something, with the setting being `PacketType` and the setting description being `Log sending or receiving packets, or both`, I think

Reply:
Added descriptions for all settings...oddly, in game some of the words in the descriptions get cut off. I'm unsure if this is something I am doing wrong, or a bug with the GUI.

Reply:
GUI